src/luascript.cpp

```diff
@@ -11912,10 +11912,10 @@ int LuaScriptInterface::luaMonsterTypeGetAttackList(lua_State* L)
 		return 1;
 	}
 
-	lua_createtable(L, monsterType->spellAttackList.size(), 0);
+	lua_createtable(L, monsterType->attackSpells.size(), 0);
 
-	int index = 0;
-	for (const auto& spellBlock : monsterType->spellAttackList) {
+	int32_t index = 0;
```

Comment:
It's not a big deal since int32_t is the same as int on most systems today, but I rather match the Lua API types where reasonable (i.e. in this case we're only using index to pass it to lua_rawseti and not in any context where it would make sense to have it as int32_t).

If you would like to update setMetatable() and other occurrences, please open a separate PR for that so that this one doesn't get cluttered with unrelated changes.